NGS2_WITNESS_Cycle1_exploratory_exp1.R

```diff
@@ -0,0 +1,216 @@
+                  "Q35_social_networks",
+                  "Q37_social_media_hours_day",
+                  "soc_dom_orient",
+                  "comm_orient_scale",
+                  "horiz_indiv",
+                  "vert_indiv",
+                  "horiz_collect",
+                  "vert_collect",
+                  "tipi_extraversion",
+                  "tipi_agreeableness",
+                  "tipi_conscientiousness",
+                  "tipi_emot_stability",
+                  "tipi_open_experiences")
+
+#Group categorical variables
+#!!!Add "time of the day", "computer settings", "Race/ethnicity", "technology access"
+
+NameListCat <- c("Q12_gender",
+                 "Q13_education",
+                 "Q14_job",
+                 "Q17_occupation",
+                 "religion",
+                 "Q24_religion_freq",
+                 "Q27_marital_status",
+                 "Q29_born_in_country",
+                 "Q22_income_feeling")
+
+empanelment$empanel_id<-empanelment$ExternalReference
+
+col.num <- which(colnames(merged.exp1.agg) %in% NameListCont)
+cont.vars <- merged.exp1.agg[,col.num]
+cont.data<-cbind(cont.vars, merged.exp1.agg$prob.cooperate)
+
+col.num <- which(colnames(empanelment) %in% NameListCat)
+cat.vars<- merged.exp1.agg[,col.num]
+cat.data<-cbind(cat.vars, merged.exp1.agg$prob.cooperate)
+
+all.vars<-c(NameListCat, NameListCont)
+
+### Correlation visualizations for continuous variables
+
+corrgram (cont.vars)
+chart.Correlation(cont.vars)
+summary(cont.vars$Q35_social_networks)
+
+### Association visualizations from categorical variables
+
+BoxPlotter<-function(var.pos) {
+fill <- "gold1"
+line <- "goldenrod2"
+var.name<-colnames(cat.vars)[var.pos]
+levels(cat.vars[,var.pos]) <- gsub(" ", "\n", levels(cat.vars[,var.pos]))
+levels(cat.vars[,var.pos]) <- gsub("/", "/\n", levels(cat.vars[,var.pos]))
+customboxplot <- ggplot(cat.vars, aes(cat.vars[, var.pos], merged.exp1.agg$prob.cooperate)) +
+  geom_boxplot(fill = fill, colour = line, notch = TRUE) +
+  xlab(paste(var.name))
+print(customboxplot)
+}
+for (i in 1:(length(NameListCat))){
+  cat.factor<-colnames(cat.vars)[i]
+  print(cat.factor)
+  BoxPlotter(i)
+  plotname<-paste(cat.factor, "_prob.cooperate.png", sep="")
+  ggsave(filename=plotname, height=7,width=9)
+  }
+
+########################################################################################################################
+# Univariate test function for predictors using logistic regression
+########################################################################################################################
+
+logit.tester <- function(ind.var) {
+  vars<-paste(c(ind.var))
+  formula.logit<-as.formula(paste("cooperate~", vars, sep=""))
+  logit.test <- glm(formula.logit, data = merged.exp1, family = "binomial")
+  logit.multiwayvcov.test <- cluster.vcov(logit.test, cbind(merged.exp1$sessionnum, merged.exp1$pid))
+  output<-coeftest(logit.test, logit.multiwayvcov.test)
+  return(output)
+}

```

Comment:
Nice; this is a good expansion on what I suggest above about function-izing the logit tests in the confirmatory analysis on experiment 1.